### ISSUE TITLE
Authentication cookies not readable on localhost

### ISSUE THREAD

**Issue by last61474** [go-pkgz/auth#67]
I have hosted my go based direct oauth login server at localhost:9999

My frontend server is running at localhost:3000, I am able to hit /oauth/local/login api from my browser successfully, but it doesn't populate cookies in any further requests.

I tried lots of solutions on google, most of them are pointing to enable credentials in header. I did try to set these paramerters in my header response:
```
		responseWriter.Header().Set("Access-Control-Expose-Headers", "XSRF-TOKEN")
		responseWriter.Header().Set("Access-Control-Allow-Origin", "localhost:9999, localhost, localhost:3000")
		responseWriter.Header().Set("Access-Control-Allow-Credentials", "true")
```

But it still doesn't work. Any idea on what I am missing here? Do I have to enable some flag in `auth` library?

PS: It's working fine if I serve frontend files from my go server directly.

**Comment by umputun** [comment]
_pls see the [working example](https://github.com/go-pkgz/auth/tree/master/_example) provided as a part of the library_

Regarding your question - I'm not sure what you trying to achieve but to me, it sounds like you want to set a cookie on localhost:9999 and read them from the localhost:3000. This type of configuration is sort of exotic and in practical use case people hide both servers behind the proxy, so both requests served on the same server:port and, from the browser point of view, came from the same server.

I don't think this issue has anything to do with auth library, but rather browser security limitations. I'd suggest to set up a proxy in front of your both :9999 and :3000 servers and likely this issue will go away/

**Comment by akellbl4** [comment]
@last61474 if I right, you are trying to set cookies with CORS.

You can try smth like this:
```
	responseWriter.Header().Set("Access-Control-Allow-Origin", "*")
	responseWriter.Header().Set("Access-Control-Allow-Credentials", "true")
	responseWriter.Header().Set("Access-Control-Allow-Methods", "GET, POST")
```

By the way, you can find docs here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Comment by last61474** [comment]
@umputun Thanks for the suggestion of reverse proxy. I kept it as last resort as this is more for local development environment. I was looking for more easier purpose.
I think cross sub-domain cookies are very common for web application. Normally organizations serve frontend and backend on separate sub domains(we are doing same using CORS) so I thought their should have been some way for handling it in localhost too. Maybe I am wrong here..

@akellbl4 I did go through mozilla link you have provided and as I mentioned in OP, I have already provided all these headers. Also browser doesn't allow you to put * for `Access-Control-Allow-Origin` if `Access-Control-Allow-Credentials` is set to true.

I have tried other solutions on google but nothing solved it for localhost. I guess for now I will use reverse proxy.

Thanks guys for your inputs. Shall I keep it open for some time to check if anyone else have some other suggestions?
